client/db/src/lib.rs

```diff
@@ -1079,6 +1081,30 @@ impl<Block: BlockT> Backend<Block> {
 		Self::new(db_setting, canonicalization_delay).expect("failed to create test-db")
 	}
 
+	/// Same function as `new_test_with_tx_storage`,just change the type of `blocks_pruning` to
+	/// `BlocksPruning`.
+	#[cfg(any(test, feature = "test-helpers"))]
+	pub fn new_test_with_tx_storage_2(
```

Comment:
Can we not just change `new_test_with_tx_storage` to take `BlocksPruning` by default?

Reply:
Hmm,,I think that will work too.